### ISSUE TITLE
Maps not loading from map server / map tiles already cached show

### ISSUE THREAD

**Issue by bic45** [microg/android_packages_apps_GmsCore#455]
I have a strange issue since an undefined period of time:

On all of my devices (Samsung Tablet with AOSP 6 ROM, Sony XZ with Android 7.1.1 or Sony Z3C with Android 7.0 apps using in app maps do not show map tiles anymore. Any POI's or markers supplied by the apps show up.

Tiles which I used in the past (undefined time scale) show up. So, cached tiles seem to be displayed but not ones that I never visitied.

I can see that there is network activity. I even see that there is a througput when I move to unloaded map tiles (50-100kbps).

This happens with build 105 and 107. Unable to tell if it already happened with older builds. For sure it worked at some point in time, otherwise I wouldn't have cached tiles that show.

Any help is appreciated.

**Comment by DissCent** [comment]
I just installed MicroG for the first time ever (using the Lineage patched ROM from the website) and although apps do not complain that play services aren't installed, map tiles are not loading at all. I can see markers on the map and geolocating is working as intended, but I am unable to see any part of the map.

**Comment by eggercomputerchaos** [comment]
Same to me for devices i9300 (S3), d855 (LG G3, jfvelte (s4 9515) --> klte (s5) maps are loaded 
and for example DriveNow App crash on start

**Comment by bic45** [comment]
@DissCent 

I can fully confirm what you say in the EDIT. Same here.

On one device the tilecache.db is about 2.33 MB in size. There is also a tilecache.db-journal which is 16.5 KB in size. This device is the one that shows me map tiles that I visited some time ago, when the loading of tiles was working.

On the other devices that don't load/show any tiles both files are 0 in size.

So you might be right in assuming that there is a problem with the caching process.

**Comment by MaferCJ** [comment]
It seems that the server responses with a "400 Bad Request. Your browser sent a request that this server could not understand"

**Comment by bic45** [comment]
@MaferCJ 

Thanks for that. Interesting find. How did you find out? I can't see any logcat entry.

**Now lets try a shot in the dark:** Is there anybody out there who can confirm that map tiles are loading? Very important before you reply: You must move to an area that you haven't visited before to make sure that it really needs to load tiles from the server.

Feedback would be much appreciated. Thanks in advance.

**Comment by eggercomputerchaos** [comment]
@bic45 
no map tiles are loaded in an area that I haven't visited before :-(

**Comment by bic45** [comment]
Thanks eggercomputerchaos!

The problem is materializing. We need more confirmations. Thanks!

**Comment by vaglan** [comment]
Same hier.No map tiles loaded.Device redmi 4a on cr_droid nougat rom

**Comment by HugoPoi** [comment]
I confirm the problem on LineageOS 14.1 with gms core 0.2.4-107, I don't have any tiles loaded on fresh install nothing appear.

**Comment by eggercomputerchaos** [comment]
It seems to be not a new issue :
https://github.com/microg/android_frameworks_mapsv1/issues/11

**Comment by DissCent** [comment]
I doubt that these issues are related because loading map tiles has worked before (as stated by @bic45 ). Also the issue you mentioned was created over one year ago and affected the maps API v1 (which has been deprecated for quite a while now).

**Comment by bic45** [comment]
Well, the point might be worth to check. To make sure that they are not interconnected we need an app that for sure uses the API v2. Any ideas?

**Comment by CharlesGayot** [comment]
Same problem here, made a full wipe on the phone, no tiles are loading. 
Seeing the comments, just downloaded an app using API v2 examples:
https://play.google.com/store/apps/details?id=pl.mg6.android.maps.extensions.demo 
The tiles are not loading, but other features seem to work.

**Comment by kolya-ay** [comment]
Same here. LinageOS 14.1 with 0.2.4-105 downloaded from the site.

**Comment by eggercomputerchaos** [comment]
@bic45 
Does the map tiles loaded in the past? Do you know which gms core version?

@DissCent 
yes, the issue was created one yer ago, but the result is the same. I asked "RalfJung", he said the issue was never fixed. 

It is sad that maps cannot be shown in some Apps which use Google-Maps.
I would appreciate if a solution could be found.
Wish you all nice days and a happy new year 2018.

**Comment by MaferCJ** [comment]
The problem is the line break on the [http header](https://github.com/microg/android_external_vtm/blob/5c825ea9eb535ecc0097e43609eb94024c5d8768/vtm/src/org/oscim/tiling/source/LwHttp.java#L98-L112). Changing the line break from `\n` to `\r\n` and it worked again.
Source: https://stackoverflow.com/a/5757349/3525001

**Comment by bic45** [comment]
**@eggercomputerchaos** 
Sorry, no. I can't say which version.

**@MaferCJ** 
Nice find, again. Now, the question is: How do you path that on 107? I am not a coder so I'm unable to compile the beast myself. Do you have a patched version?

**Comment by yeahchaos** [comment]
I already built a patched packet right after MaferCJ's hint - and it
works!

Until there is an official build, you can download mine here:
https://www.dropbox.com/sh/rsxrg7xgqfkvvz1/AADbOi9VcgBYy4r3B9zA4yqUa?dl=0

**Comment by bic45** [comment]
**@yeahchaos** 
Could it be that you linked the wrong directory? There is only one microg.apk in there but its way to small. In addition it looks like there a some private files which probably shouldn't be public...

But thanks already for the offer! Very kind of you.

**Comment by yeahchaos** [comment]
@bic45 
You're right, thanks!
Now the link is correct!

**Comment by bic45** [comment]
@yeahchaos: Thanks a lot! 

Ok, here we are, there it is! The microg-107 built by yeahchaos works. Now, the question is: What needs to be done to integrate this into the next official build? Also marv-i-in should be informed. I don't know if he reads this thread. Has anybody got contact with him?

**Comment by JonnyTech** [comment]
@yeahchaos, good work, I recommend that you submit a pull request to get the attention of @mar-v-in

**Comment by bic45** [comment]
As I wrote, I am not a coder. Sorry. No idea how this works. @yeahchaos could you be so kind, please?

**Comment by yeahchaos** [comment]
I will. But not until tomorrow,  since I am already with my family, celebrating x-mas...

**Comment by vaglan** [comment]
@yeahchaos Works for me .Thaaanks. Good work

**Comment by eggercomputerchaos** [comment]
@yeahchaos , @MaferCJ and all other poeple  :  Also, thanks so much from me.

just as bic45 i am not a dev and don't know how does it works? It would be nice to get a short explanation what to do. Thanks

Does anybody already pull a request to mar-v-in ?

**Comment by Freeedim** [comment]
Great guys!
Thanks for the work done!
But how do you update the gms core with the provided apk? My phone tells me that there is a conflicting package, even if I uninstall the regular version with Titanium backup (my microg suite was flashed as system apps). I guess it is a signature issue, but I have no clue about how to manage this.

**Comment by mar-v-in** [comment]
I will update the signed build tomorrow

**Comment by JonnyTech** [comment]
@mar-v-in does the newly released build (0.2.4-108) contain the fix, as I cannot find a commit that references it?

**Comment by ArchangeGabriel** [comment]
@JonnyTech It is hidden in the `extern/vtm` update of https://github.com/microg/android_packages_apps_GmsCore/commit/464d45d552eb7bf075fd19ea1a5e9d8b8e7078b4, which is quite huge (428 commits! https://github.com/microg/android_external_vtm/compare/b651e095f019649de2adc578264c8d582551f657...a47215d9848d5f0ebd1be7e281b48e531026a47f).

**Comment by JonnyTech** [comment]
@ArchangeGabriel thanks, I found [this commit](https://github.com/microg/android_external_vtm/commit/9977061680447e2ffa6a9e01a8096f61a9f2ab93) which was also referenced above - I did not realise that android_packages_apps_GmsCore pulled in files from android_external_vtm

**Comment by mar-v-in** [comment]
As already pointed out, 0.2.4-108 contains the required fix (as well as several performance and stability improvements in map rendering)

**Comment by ArchangeGabriel** [comment]
Thank you very much @mar-v-in for your continued work on this project! :D

Hope to see nice improvements for μG in 2018. ;)
